### PR TITLE
Implement peer ejection.

### PR DESCRIPTION
**Pull request by taphu**

### CODE REVIEW

File: src/Network/Legion/Conduit.hs [hunk 1]
```diff
@@ -25,13 +25,7 @@ chanToSource chan = forever $ yield =<< liftIO (readChan chan)
  Convert an chanel into a Sink.
```
Comment (MHova) [inline src/Network/Legion/Conduit.hs]: "Convert **a** **channel**"?

Comment (taphu) [inline src/Network/Legion/Conduit.hs]: yes.

Comment (taphu) [inline src/Network/Legion/Conduit.hs]: fixed in e9e97ce71777ff61efe359f51820f30235dc2f65

File: src/Network/Legion/PowerState.hs [hunk 2]
```diff
@@ -178,6 +178,33 @@ mergeEither (PowerState o1 i1 d1) (PowerState o2 i2 d2) | o1 == o2 =
...
+      broadcasting updates.
+
+      In reality, this will probably always be a no-op because the
+      message dispatcher in the main state machine will immediately
+      drop messages that originate from unknown peers (where "unknown"
+      includes peers that have been ejected), so it is unlikely that any
+      renegade merge requests will make it this far, but you can never
+      be too paranoid I guess.
+    -}
+    removeRenegade ps =
```
Comment (MHova) [inline src/Network/Legion/PowerState.hs]: Pattern match/pun `deltas`?

File: src/Network/Legion/Conduit.hs [hunk 3]
```diff
@@ -11,27 +11,21 @@ import Control.Concurrent (forkIO)
...
-  Convert a chanel into a Source.
+  Convert a channel into a Source.
 -}
 chanToSource :: (MonadIO io) => Chan a -> Source io a
 chanToSource chan = forever $ yield =<< liftIO (readChan chan)
 
 
 {- |
- Convert an chanel into a Sink.
+ Convert an channel into a Sink.
```
Comment (MHova) [inline src/Network/Legion/Conduit.hs]: "**a** channel" :neckbeard:

File: src/Network/Legion/Conduit.hs [hunk 4]
```diff
@@ -15,14 +15,14 @@ import Data.Conduit (Source, Sink, ($$), ($=), yield, awaitForever)
...
```
Comment (glasserc) [inline src/Network/Legion/Conduit.hs]: So close and yet so far! *a* channel, not *an* channel.

**Comment by MHova** [comment]
Otherwise :+1: 

At some point I'm going to fix all the typos sprinkled everywhere...

**Pull request merged**